mmu-to-hats/main.py

```diff
@@ -87,12 +91,20 @@ def read(self, input_file: str, read_columns: list[str] | None = None):
             n_rows = h5_file[read_columns[0]].shape[0]
             chunk_size = max(1, n_rows // num_chunks)
             for i in range(0, n_rows, chunk_size):
-                data = {
-                    col: np_to_pyarrow_array(h5_file[col][i : i + chunk_size])
-                    for col in read_columns
-                }
-                table = pa.table(data)
-                yield table
+                if set(read_columns) == set(["ra", "dec"]):
+                    data = {
+                        col: np_to_pyarrow_array(h5_file[col][i : i + chunk_size])
+                        for col in read_columns
+                    }
+                    table = pa.table(data)
+                    yield table
+                else:
+                    data = {
+                        col: h5_file[col][i : i + chunk_size]
+                        for col in read_columns
+                    }
+                    table = self.transform_klass.dataset_to_table(data)
+                    yield table
```

Comment:
I'm not sure I understand why we need both options here. Also, for the coordinate transformation, just col: `h5_file[col][i : i + chunk_size]` should be fine, so we can drop `np_to_pyarrow_array`

Reply:
we go over this function multiple times (first just to extract the coordinates and then figure out what partitioning we need and which objects belong to which partition, then the second time to read the data and create the partitions accordingly) and I can either do the distinction here or in the dataset_to_table method. To keep the dataset_to_table as simple as possible I chose to do this in here.